### PR TITLE
feat(cb2-6092): customise emissions sections

### PR DESCRIPTION
**Pull request by CormacBJSS**
## Ticket title

Customise Emissions section / certificate fields based on test result
[https://dvsa.atlassian.net/browse/CB2-6092]

## Checklist

- [ ] Branch is rebased against the latest develop/common
- [ ] Necessary `id` required prepended with `"test-"` have been checked with automation testers and added
- [ ] Code and UI has been tested manually after the additional changes
- [ ] PR title includes the JIRA ticket number
- [ ] Squashed commits contain the JIRA ticket number
- [ ] Link to the PR added to the repo
- [ ] Delete branch after merge

### CODE REVIEW

**Review by leifkemp-bjss — COMMENTED** [review]
Could probably do with unit tests for the validator and error messages

**Review by kpearmanbjss — DISMISSED** [review]

File: src/app/forms/validators/custom-async-validators.ts [hunk 1]
```diff
@@ -70,15 +70,16 @@ export class CustomAsyncValidators {
     };
   }
 
-  static requiredIfNotResult(store: Store<State>, result: resultOfTestEnum): AsyncValidatorFn {
+  static requiredIfNotResult(store: Store<State>, result: any): AsyncValidatorFn {
```
Comment (gjulien-bjss) [inline src/app/forms/validators/custom-async-validators.ts:73]:
```suggestion
  static requiredIfNotResult(store: Store<State>, result: resultOfTestEum | resultOfTestEnum[]): AsyncValidatorFn {
```

File: src/app/forms/validators/custom-async-validators.ts [hunk 2]
```diff
@@ -92,4 +93,28 @@ export class CustomAsyncValidators {
...
+            if ((Array.isArray(result) ? !result.includes(currentResult) : currentResult !== result) && newValue && (control.value === null || control.value === undefined || control.value === '')) {
+              return { requiredIfNotResultAndSiblingEquals: { sibling: siblingControl.meta.label } };
+            }
+          }
+
+          return null;
+        })
+      );
+  }
+
```
Comment (gjulien-bjss) [inline src/app/forms/validators/custom-async-validators.ts:119]: Why can you not attach the `requiredIfSiblingEquals` validator (or create it if it doesn't exist) and the `requiredIfResult` validators on the control?

Comment (CormacBJSS) [reply src/app/forms/validators/custom-async-validators.ts:119]: What I am trying to achieve is rules like: "particulateTrapFitted should only be required if modType code is p and the result is pass"

The problem is the validators are independent of each other so there is no way to do an "AND" condition as far as I know. If either validator fails then the error message is shown. Is there a way around that?

File: src/app/forms/validators/custom-async-validators.ts (hunk 1 above)
Comment (CormacBJSS) [reply src/app/forms/validators/custom-async-validators.ts:73]: That looks good thanks.

**Review by kpearmanbjss — APPROVED** [review]

**Review by mcadenas-bjss — APPROVED** [review]

**Pull request merged**